### PR TITLE
Run 20 iterations by direct invocation

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Refactor test scripts to run the tests with 20 iterations in a loop internally as before, and also 20 times in a loop externally.

### CODE REVIEW

**Comment by sbesson** [comment]
From https://ci.openmicroscopy.org/view/Experimental/job/OME-FILES-performance-win/132/console

```
( was unexpected at this time.
Build step 'Execute Windows batch command' marked build as failure
```

File: scripts/run_benchmarking [hunk 1]
```diff
@@ -2,38 +2,77 @@
 set -e
 set -x
 
+datapath=/data/images/ptest
```
Comment (sbesson) [inline scripts/run_benchmarking]: This should be reverted in favor of the line below to be executed in our current production system

File: scripts/run_benchmarking [hunk 2]
```diff
@@ -2,38 +2,77 @@
...
+#datapath="/data"
+binpath=/tmp/inst
+#binpath="/install/bin"
```
Comment (sbesson) [inline scripts/run_benchmarking]: Same as above

**Comment by rleigh-codelibre** [comment]
Oops, yes, that was for local testing.

File: scripts/run_benchmarking [hunk 3]
```diff
@@ -2,38 +2,75 @@
...
+        cd $BF_JACE_HOME
+        # Can throw boost: mutex exception
+        ${binpath}/metadata-performance-jace ${iterations} "$input" ${outpath}/${test}-jace.ome.xml ${resultpath}/${test}-metadata-linux-jace.tsv || true
+        for i in $(seq ${iterations}); do
+            ${binpath}/metadata-performance-jace 1 "$input" ${outpath}/${test}-jace-${i}.ome.xml ${resultpath}/${test}-metadata-linux-jace-${i}.tsv || true
+        done
+
+        jace_iterations=${iterations}
+        # NB: takes ~1h30 to complete
+        if [ "$test" == "BBBC" ]; then
```
Comment (sbesson) [inline scripts/run_benchmarking]: Should this be `bbbc` (lowercase) instead? At the moment, JACE is running 20 executions for the BBBC plate on beluga

**Review by sbesson — COMMENTED** [review]
One issue with the number of JACE executions logic. Otherwise, https://ci.openmicroscopy.orgjob/OME-FILES-performance-win/135/ has run successfully  using the latest state of this PR and the generated results are  available for inspection in the archived artifacts.

In terms of generated content and disk space, each iteration is now producing a copy using the index as a suffix in the filename. Although keeping a copy of the generated output per component has been useful for comparing, debugging & checking data integrity, I don't think there is value in keeping 21 identical copies of each OME-TIFF/OME-XML. I would suggest we overwrite the same copy for each `{bbbc,mitocheck,tubhiswt}-{cpp,java,jace}.ome.{tiff,xml}` combination. I can open a follow-up PR to address those or we can push additional commits here as you prefer.

**Comment by rleigh-codelibre** [comment]
@sbesson Fixes pushed for both problems.

**Review by sbesson — APPROVED** [review]
https://ci.openmicroscopy.org/view/Experimental/job/OME-FILES-performance-win/136/console looks good. The Linux benchmarking suite is also running and everything behaves as expected.

**Comment by sbesson** [comment]
After 18h of execution (including 15h for the JACE/BBBC tests), everything works as expected and this is ready to merge. Uploading the
[Linux results](https://github.com/openmicroscopy/ome-files-performance/files/786933/results-17-02-2017.tar.gz) and
[Windows results](https://github.com/openmicroscopy/ome-files-performance/files/786934/archive.zip)
to this PR comment but I can also push them to a branch under version control is that makes them easier to retrieve @rleigh-codelibre. Merging.

**Pull request merged**